nni/algorithms/compression/v2/pytorch/base/scheduler.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import List, Dict, Tuple
+
+from torch.nn import Module
+from torch.tensor import Tensor
+
+
+class BasePruningScheduler:
+    def generate_task(self) -> Tuple[int, Module, List[Dict], Dict[str, Dict[str, Tensor]]]:
+        """
+        Returns
+        -------
+        Tuple[int, Module, List[Dict], Dict[str, Dict[str, Tensor]]]
+            Return new task id, model under pruning, config list used in this task and pre-masks.
+        """
+        raise NotImplementedError()
+
+    def record_task_result(self, task_id: int, pruned_model: Module, masks: Dict[str, Dict[str, Tensor]], score: float,
+                           up_model_masks: Dict[str, Dict[str, Tensor]]):
+        """
+        Used to record the task result.
+
+        Parameters
+        ----------
+        task_id
+            The id of the finished task.
+        pruned_model
+            The pruned model after `pruning_one_step`.
+        masks
```

Comment:
Please clarify the difference between the `masks` and the `up_model_masks`?

Reply:
add docstring, please refer to `TaskResult` in `scheduler.py`